File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        if list1 is None and list2 is None:
            return None
            
        current1 = list1
        current2 = list2
        answer = ListNode()
        temp = answer
        while current1 != None or current2 != None:
            temp.next = ListNode()
            temp = temp.next
            if current1 is None:
                temp.val = current2.val
                current2 = current2.next            
                continue
            elif current2 is None:
                temp.val = current1.val
                current1 = current1.next
                continue
            
            if current1.val < current2.val:
                temp.val = current1.val
                current1 = current1.next
            else:
                temp.val = current2.val
                current2 = current2.next
        
        return answer.next
                

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode curr1 = list1;
        ListNode curr2 = list2;
        ListNode answerHead = new ListNode(0, null);
        ListNode answerCurr = answerHead;
        while (curr1 != null || curr2 != null){
            if (curr1 == null){
                answerCurr.next = curr2;
                curr2 = curr2.next;
                answerCurr = answerCurr.next;
                continue;
            }
            if (curr2 == null){
                answerCurr.next = curr1;
                curr1 = curr1.next;
                answerCurr = answerCurr.next;
                continue;
            }

            if (curr1.val > curr2.val){
                answerCurr.next = curr2;
                curr2 = curr2.next;    
            } 
            else {
                answerCurr.next = curr1;
                curr1 = curr1.next;    
            }
            answerCurr = answerCurr.next;
        }

        return answerHead.next;
    }
}